### PR TITLE
Add new allowWindowsEscape option

### PR DESCRIPTION
**Pull request by jdzak**
The `allowWindowsEscape` option was just added to glob 8.1. https://github.com/isaacs/node-glob/blob/main/changelog.md#81

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/glob/index.d.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ declare namespace G {
         nonegate?: boolean | undefined;
         nocomment?: boolean | undefined;
         absolute?: boolean | undefined;
+        allowWindowsEscape?: boolean | undefined;
```
Comment (peterblazejewicz) [inline types/glob/index.d.ts:59]: please bump the version in the header, otherwise lgtm

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
version bump is a must

**Comment by jdzak** [comment]
@peterblazejewicz Did I bump the version in the header correctly?

File: types/glob/index.d.ts (hunk 1 above)
Comment (jdzak) [reply types/glob/index.d.ts:59]: Done. TY!

**Review by peterblazejewicz — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@jdzak thanks!

**Comment by jdzak** [comment]
Ready to merge

**Pull request merged**